### DISCUSSION TITLE
Scalability - simple tasks are very slow

### DISCUSSION THREAD

**Discussion by mvanelst** [bluebanquise/bluebanquise#946] [Cluster deployment]
Hello,

this is my first attempt on using bluebanquise (the current 3.0 version). The intent is to use it to orchestrate an existing large cluster.

I've tried to run simple roles like hosts_file with a test inventory of only about 300 nodes, which took (on an average 2.7GHZ Skylake CPU, SLES15, python3.11) about 4 minutes (CPU time) on the management node itself before even deploying the computed template.

I found that almost all the CPU time went into calculating j2_bb_nodes, and this mostly from fetching j2_icebergs_main_network_dict for every host in the bb core plugin. And that when I haven't configured the bb_icebergs_system yet, so the dict is always empty.

Is that something to be expected?

**Comment by oxedions** [comment]
Dear @mvanelst 

My apologies for this huge delay in my answer.
First, Github strangely do not notify me for new threads in Discussions. Secondly, I was really over loaded recently.

I agree that the current j2_bb_nodes is not properly made, as iceberg heavy call to j2_icebergs_main_network_dict should not be triggered in case of non icebergs cluster.

I tried to reproduce this behavior, without success for now:

I created this small script:

```bash
echo "all:" > nodes_c.yml
echo "  hosts:" >> nodes_c.yml
for (( i=1; i<=500; i++ ))
do
cat << EOF >> nodes_c.yml
    c$i:
      bmc:
        name: bc$i
        ip4: 10.10.103.$i
        network: net-admin
        mac: 2a:2b:3c:2d:5e:$i
      network_interfaces:
        - interface: enp1s0
          ip4: 10.10.3.$i
          mac: 1a:2b:3c:4d:5e:$i
          network: net-admin
        - interface: ib0
          ip4: 10.20.3.$i
          mac: 1a:2b:3c:4d:5e:$i
          network: ic

EOF
done
```

To generate 500 nodes in my inventory, and I added these nodes into hw, os and fn groups, and I also added relevant networks into the inventory.

On my laptop, which has an i5-4200U CPU, using ansible-core 2.17 the time to generate and print the j2_bb_nodes was 19.68s.

Do you have something inside your inventory that could have created an unexpected loop ? :slightly_smiling_face: 

In the meantime, I will update the core to reduce calculation load.

**Reply by oxedions** [reply]
Removing the icebergs related part reduce time to 3.66s, which is a HUGE improvement. I will change the way value is calculated to improve this.
